apps/native-component-list/src/screens/WebBrowser/OpenAuthSessionAsyncDemo.tsx

```diff
@@ -1,60 +1,41 @@
 import * as Linking from 'expo-linking';
 import * as WebBrowser from 'expo-web-browser';
 import React from 'react';
-import { StyleSheet } from 'react-native';
 
-import Button from '../../components/Button';
-import Configurator, { ConfiguratorChoiceType } from '../../components/Configurator';
-import HeadingText from '../../components/HeadingText';
-import MonoText from '../../components/MonoText';
-
-const openAuthSessionConfigurationChoices: ConfiguratorChoiceType[] = [
-  {
-    name: 'shouldPrompt',
-    title: 'Should prompt',
-    initial: false,
-  },
-  {
-    name: 'createTask',
-    title: 'Create task',
-    initial: true,
-  },
-];
+import FunctionDemo from '../../components/FunctionDemo';
 
 export default function OpenAuthSessionAsyncDemo() {
-  const [options, setOptions] = React.useState<{ shouldPrompt?: boolean; createTask?: boolean }>(
-    {}
-  );
-  const [authResult, setAuthResult] = React.useState<WebBrowser.WebBrowserAuthSessionResult | null>(
-    null
-  );
-
-  const openAuthSession = React.useCallback(async () => {
-    const redirectUrl = Linking.makeUrl('redirect');
-    const result = await WebBrowser.openAuthSessionAsync(
-      `https://fake-auth.netlify.com?state=faker&redirect_uri=${encodeURIComponent(
-        redirectUrl
-      )}&prompt=${options.shouldPrompt ? 'consent' : 'none'}`,
-      redirectUrl,
-      { createTask: options.createTask }
-    );
-
-    setAuthResult(result);
-  }, [options]);
+  const redirectUrl = Linking.createURL('redirect');
 
   return (
-    <>
-      <HeadingText>openAuthSessionAsync</HeadingText>
-      <Configurator choices={openAuthSessionConfigurationChoices} onChange={setOptions} />
-      <Button style={styles.button} onPress={openAuthSession} title="Open web auth session" />
-      {authResult && <MonoText>result: {JSON.stringify(authResult, null, 2)}</MonoText>}
-    </>
+    <FunctionDemo
+      namespace="WebBrowser"
+      name="openAuthSessionAsync"
+      parameters={[
+        {
+          name: 'url',
+          type: 'constant',
+          value: 'url',
+        },
+        { name: 'redirectUrl', type: 'constant', value: redirectUrl },
+        {
+          name: 'options',
+          type: 'object',
+          properties: [{ name: 'createTask', type: 'boolean', initial: true }],
+        },
+      ]}
+      additionalParameters={[{ name: 'shouldPrompt', type: 'boolean', initial: false }]}
+      actions={(
+        _: string,
+        redirectUrl: string,
+        options: WebBrowser.WebBrowserOpenOptions,
+        shouldPrompt: boolean
+      ) => {
+        const url = `https://fake-auth.netlify.com?state=faker&redirect_uri=${encodeURIComponent(
+          redirectUrl
+        )}&prompt=${shouldPrompt ? 'consent' : 'none'}`;
+        return WebBrowser.openAuthSessionAsync(url, redirectUrl, options);
+      }}
```

Comment:
Why do you put so many things into the render function? It becomes harder to read, less concise and less efficient.

Reply:
Yeah! I've refactored and cleaned up the whole PR now, so give it another shot! 😁